realm/src/main/java/io/realm/RealmQuery.java

```diff
@@ -255,9 +261,13 @@ private boolean containsDot(String s) {
      * field type
      * @throws java.lang.RuntimeException Any other error
      */
-    public RealmQuery<E> equalTo(String fieldName, long value) {
+    public RealmQuery<E> equalTo(String fieldName, Short value) {
```

Comment:
@cmelchior User might get a `null` value from the last query, then chain the result (`null`) in the next query. In this case, it would be really convenient to take `null` as the input for the `equal` method. and also, `equalTo("someType", null) won't compile, which makes user think when calling it by casting to some type. I prefer to have it.